.github/workflows/linter.yml

```diff
@@ -0,0 +1,61 @@
+name: Lint/Sec Checks
+
+on: 
+  workflow_dispatch:
+  push:
+    paths-ignore:
+      - '**.md'
+  pull_request:
+    branches: [main]
+
+jobs:
+  go-linters:
+    name: Go linters
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Check code formatting
+        uses: golangci/golangci-lint-action@v3
+        with:
+          # https://golangci-lint.run/usage/linters/
+          args: --disable-all --enable goimports
+
+      - name: Check code complexity
+        uses: golangci/golangci-lint-action@v3
+        with:
+          args: --disable-all --enable funlen --enable gocyclo 
```

Comment:
Thanks for suggesting these changes, Rafid. Can we please just scale it down a bit so we can get started. Specifically, for linters I'd suggest having: gofmt (not goimports), errcheck, staticcheck, unused.

Reply:
Updated in https://github.com/canonical/chisel/pull/31/commits/c013935a13f0b2c47203777501ea88610b703d3f. Thank you, there were quite a lot of linters indeed.